graylog2-server/src/main/java/org/graylog2/database/MongoDBVersionCheck.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.database;
+
+import com.github.zafarkhaja.semver.Version;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoException;
+import com.mongodb.client.MongoDatabase;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.graylog2.bootstrap.preflight.PreflightCheckException;
+import org.graylog2.shared.utilities.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+public class MongoDBVersionCheck {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBVersionCheck.class);
+    private static final Version MINIMUM_MONGODB_VERSION = Version.forIntegers(2, 4);
```

Comment:
The minimum MongoDB version is 3.6 or even 4.0. Our documentation says 3.6.

Reply:
I took the check from existing code. But sure, I can adopt it.